src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -221,6 +232,8 @@ export class NotebookServerProvider implements IJupyterServerProvider {
             uri: serverURI,
             skipUsingDefaultConfig: !useDefaultConfig,
             purpose: Identifiers.HistoryPurpose,
+            kernelConnection: options.kernelConnection,
+            metadata: options.metadata,
```

Comment:
We need to pass the `metadata` for WebView notebooks (this was old code that I happend to have removed)
We need to pass `KernelConnectionMetadata` as in NativeNotebooks we search & preselect a kernel when a notebook is opened, hence we want to pass the pre-selected one to avoid searching all over again & using something else (this is what was partially fixed in another PR, but this code path was missing - non-raw when opening an existing nb).